src/renderer/views/wallet/AssetDetailsView.tsx

```diff
@@ -26,14 +41,33 @@ export const AssetDetailsView: React.FC = (): JSX.Element => {
     resetTxsPage
   } = useWalletContext()
 
-  const { asset, walletAddress } = useParams<AssetDetailsParams>()
-  const oSelectedAsset = useMemo(() => O.fromNullable(assetFromString(asset)), [asset])
-  const oWalletAddress = useMemo(() => O.fromNullable(walletAddress || undefined), [walletAddress])
+  const {
+    service: {
+      pools: { poolAddresses$ }
+    }
+  } = useMidgardContext()
 
-  // Set selected asset once
-  // Needed to get all data for this asset (transactions etc.)
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  useEffect(() => setSelectedAsset(oSelectedAsset), [])
+  const [bnbPoolAddress] = useObservableState(
+    () =>
+      FP.pipe(
+        oSelectedAsset,
+        // We do need bnb pool address for BNB.RUNE assets only
+        O.filter(isRuneBnbAsset),
+        O.fold(
+          // No subscription of `poolAddresses$ ` needed for other assets than BNB.RUNE
+          () => Rx.of(O.none),
```

Comment:
with this info,  proposed changes to the `getPoolAddressByChain` (to make it work based on `Asset` instead of `Chain`) AND currying that function as:

```ts
const getPoolAddressByAsset= (oAsset: O.Option<Asset>) => (
  endpoints: ThorchainEndpoints
): O.Option<PoolAddress> =>
  FP.pipe(
    oAsset,
    O.map(({ chain }) => chain),
    O.chain((chain) =>
      FP.pipe(
        endpoints,
        A.findFirstMap((endpoint) => (endpoint.chain === chain ? O.fromNullable(endpoint.address) : O.none))
      )
    ),
  )
  ```
  
  we can use it in really simple and readable way as

 ```ts
const [poolAddress] = useObservableState(
    () =>
      FP.pipe(
        poolAddresses$,
        liveData.map(getPoolAddressByAsset(oSelectedAsset)),
        RxOp.map(FP.flow(RD.toOption, O.flatten))
      ),
    O.none
  )
 ```

Reply:
> in general (if im not mistaken) it will not make sense in the nearest future. we will need a pool address from ANY asset's chain. is not it ?

Not in this case. `AssetDetails` does need BNB address only to provide upgrade feature for RUNE.BNB only...